megatron/training/arguments.py

```diff
@@ -1206,6 +1206,11 @@ def validate_args(args, defaults={}):
             args.recompute_granularity != 'full'
         ), 'recompute_granularity must not be full when CUDA Graphs are enabled.'
 
+    if args.use_inference_optimized_layers:
```

Comment:
Does these NVShmem kernels not provide any benefit for pretraining too? Or are we just missing backward and other kernel ops?